api/src/test/java/org/openmrs/api/PatientServiceTest.java

```diff
@@ -1848,6 +1852,56 @@ public void savePatient_shouldUpdateAnExistingPatient() throws Exception {
 		assertTrue("The gender should be new", patient2.getGender().equals("F"));
 	}
 	
+	@Test
+	public void savePatient_shouldFailSaveIfPatientsWithSameIdentifierAreSaved() {
+	    isSavePatientThreadsafe = new AtomicBoolean(false);
+		UserContext ctx = Context.getUserContext();
+		
+		Patient patient1 = new Patient();
+		patient1.setGender("M");
+		patient1.addName(new PersonName("Test1", "Sam", "Test"));
+		
+		Patient patient2 = new Patient();
+		patient2.setGender("M");
+		patient2.addName(new PersonName("Test2", "Samuel", "Test"));
+		
+		Runnable r1 = () -> savePatientOnThread(ctx, patient1);
+		       
+		Runnable r2 = () -> savePatientOnThread(ctx, patient2);
+				
+		Thread t1 = new Thread(r1);
+		Thread t2 = new Thread(r2);
+		
+		t1.start();
+		t2.start();
+		
+		try {
+			t1.join();
+		} catch (InterruptedException e) {
+			//Just Ignore
+		}	
+		
+		assertTrue("IdentifierNotUniqueException not thrown, method not Threadsafe", isSavePatientThreadsafe.get());
+		
+		//Cleans up the unrolled back Thread transactions.
+		deleteAllData();
+	}
+	public void savePatientOnThread(UserContext ctx, Patient patient) {
+		Context.setUserContext(ctx);
+        Context.openSessionWithCurrentUser();
```

Comment:
Can you also fix the above indention?

Reply:
I wonder, but locally, the indention is fine. Just to prove, hit view full file and check.

I have highlighted the it here https://github.com/samuelmale/openmrs-core/blob/fd2aed2031b647f046562da4142851669c3c4061/api/src/test/java/org/openmrs/api/PatientServiceTest.java#L1890-L1894

Reply:
If this is the unaddressed comment, I'm pretty sure my indention is fine. Have you looked at it as a file. In otherwise have looked at the above link? @dkayiwa

Reply:
If this is the unaddressed comment, I'm pretty sure my indention is fine. Have you looked at it as a file. In otherwise have looked at the above link? @dkayiwa